pkg/dump/writer/file_writer.go

```diff
@@ -49,51 +50,49 @@ func (f *FileWriter) WorkerNumber() int {
 	return FileWriterWorkerNumber
 }
 
-func (f *FileWriter) Write(ctx context.Context, data []byte, filePath string) error {
-	span, _ := tracer.StartSpanFromContext(ctx, span.DumperWriterWrite, tracer.Measured())
-	span.SetTag(tag.DumperFilePathTag, filePath)
-	span.SetTag(tag.DumperWriterTypeTag, FileTypeTag)
-	defer span.Finish()
-	filePath = path.Join(f.directoryOutput, filePath)
+// Write function writes the Kubernetes object to a buffer
+// All buffer are stored in a map which is flushed at the end of every type processed
+func (f *FileWriter) Write(ctx context.Context, k8sObj any, pathObj string) error {
+	log.I.Debugf("Writing to file %s", pathObj)
+	f.mu.Lock()
+	defer f.mu.Unlock()
 
-	buffer, ok := f.buffers[filePath]
-	if !ok {
-		// Create directories if they do not exist
-		err := os.MkdirAll(filepath.Dir(filePath), WriterDirChmod)
-		if err != nil {
-			return fmt.Errorf("failed to create directories: %w", err)
-		}
+	data, err := marshalK8sObj(k8sObj)
+	if err != nil {
+		return err
+	}
 
-		file, err := os.OpenFile(filePath, os.O_APPEND|os.O_WRONLY|os.O_CREATE, FileWriterChmod)
-		if err != nil {
-			return fmt.Errorf("failed to open file: %w", err)
-		}
-		f.files[filePath] = file
-		buffer = bufio.NewWriter(file)
-		f.buffers[filePath] = buffer
+	// Constructing output complete path to write the file
+	filePath := path.Join(f.directoryOutput, pathObj)
+
+	// Create directories if they do not exist
+	err = os.MkdirAll(filepath.Dir(filePath), WriterDirChmod)
```

Comment:
pedantic nit: I didn't catch that before before but, I guess `WriterDirChmod` should be `WriterDirMod` (you'r not **ch**anging **mod**) :D